Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -422,33 +426,115 @@ size_t SourceBufferPrivate::platformEvictionThreshold() const
     return 0;
 }
 
+Ref<GenericPromise> SourceBufferPrivate::setMaximumBufferSize(size_t size)
+{
+    if (size != m_evictionData.maximumBufferSize) {
+        m_evictionData.maximumBufferSize = size;
+        computeEvictionData(true);
+    }
+    return GenericPromise::createAndResolve();
+}
+
+void SourceBufferPrivate::computeEvictionData(bool forceNotification)
+{
+    auto evictionData = m_evictionData;
+    m_evictionData.numMediaSamples = [&]() -> size_t {
+        const size_t evictionThreshold = platformEvictionThreshold();
+        if (!evictionThreshold)
+            return 0;
+        size_t currentSize = 0;
+        iterateTrackBuffers([&](auto& trackBuffer) {
+            currentSize += trackBuffer.samples().size();
+        });
+        return currentSize;
+    }();
+    m_evictionData.contentSize = totalTrackBufferSizeInBytes();
+    m_evictionData.evictableSize = [&]() -> int64_t {
+        RefPtr mediaSource = m_mediaSource.get();
+        if (!mediaSource)
+            return 0;
+        size_t evictableSize = 0;
+        auto currentTime = mediaSource->currentTime();
+        auto timeChunkAsMilliseconds = evictionAlgorithmTimeChunkLowThreshold;
+        const auto timeChunk = MediaTime(timeChunkAsMilliseconds, 1000);
+        const auto maximumRangeEnd = std::min(currentTime - timeChunk, findPreviousSyncSamplePresentationTime(currentTime));
+
+        auto rangeStart = minimumBufferedTime();
+        auto rangeEnd = std::min(rangeStart, maximumRangeEnd);
```

Comment:
This lambda is a bit difficult to parse.
Reusing rangeStart and rangeEnd all other the place does not help.
Finding more accurate names may help.

Reply:
that's how the code in the `evictFrame` reads. While it's not great I agree, having a consistency between the eviction code itself and the estimation of the available free-able data is also good to have.

Reply:
changed the rangeStart/rangeEnd to be more meaningful in their context (before/after currentTime)